Automattic/jetpack
# Refactor registration of experimental blocks

See https://github.com/Automattic/jetpack/issues/32602

## Proposed changes:

Registers the experimental blocks (_AI Image, _AI Paragraph_) by specifying the path to their block.json file. This was made possible by https://github.com/Automattic/jetpack/pull/32697. The goal is to avoid the duplication of block metadata and use block.json as the source of truth.

In more details:
- For each block (when needed):
  - Remove the `editorScript` property from `block.json` (automatically added during the build)
  - Convert icon from dashicon to SVG in the `icon` property of `block.json`
  - Remove `icon.js`
  - Import icon helpers from the `shared-extension-utils` package
  - Register the block client and server sides with `block.json`
  - Move data from `index.js` to `block.json` and `editor.js` and delete the file
  - Move data from `attributes.js` to `block.json` and delete the file

### Other information:


## Jetpack product discussion


pedMtX-RS-p2

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






Experimental blocks are not shown in the block inserter. Below, I explain how to display them in your local test site so that they can be tested. There may be easier ways but I'm not aware of them.
- Download this branch.
- Make sure experimental blocks are enabled by adding `define( 'JETPACK_BLOCKS_VARIATION', 'experimental' );` to `/projects/jetpack/tools/docker/wordpress/wp-config.php`.
- Delete  the `supports.inserter` property in `block.json` for for the _AI Paragraph_ and _AI Image_ blocks.
- Comment out the conditions in `register_block` in `ai-image.php` and `ai-paragraph.php`
- Rebuild the extensions from `project/plugins/jetpack`: `pnpm build-extensions`
- You should now be able to see these two blocks locally in the block inserter
<img width="350" alt="Screenshot 2023-09-11 at 3 11 12 PM" src="https://github.com/Automattic/jetpack/assets/1620183/6f88d5aa-85e6-4fa5-ab46-ac0609f32420">


Test that the blocks behave as in trunk. This PR shouldn't change anything from a user's standpoint.
